### PR TITLE
fix(ara): store captured I/O on CPU for multi-GPU robustness

### PR DESCRIPTION
**Pull request by joninco**
## Summary

Extends d79a443 — that commit correctly moves I/O tensors to the weight matrix's device before L-BFGS optimization, but the captured tensors remain on their original GPU between trials. When `reset_model()` reloads the model, device assignments can change, leaving orphaned tensors on GPUs that now need that VRAM for the reloaded weights.

This adds `.cpu()` at capture time in `get_module_io`, so tensors are stored device-agnostic:

```python
input = inputs[0][:, -1, :].detach().clone().cpu()
output = outputs[:, -1, :].detach().clone().cpu()
```

This ensures:
- Zero VRAM wasted on stale device assignments between trials
- Clean CPU→target transfer regardless of how devices shuffle on reload
- No overhead on single-GPU (`.cpu()` is a no-op when already on CPU, and `.to(device)` in `ara_abliterate` handles final placement)

## Context

Hit this while running ARA on 8× RTX PRO 6000 with `device_map="auto"`. The `.to(matrix.device)` fix in d79a443 prevents the crash, but without CPU storage the captured tensors pin VRAM on devices that may get reassigned after model reload — on tight VRAM budgets this can cause OOM on subsequent trials.

## Test plan
- [x] Verified on single-GPU (`.cpu()` is no-op, `.to(device)` handles placement)
- [ ] Multi-GPU stress test with tight VRAM budget across trials

### CODE REVIEW

**Comment by p-e-w** [comment]
Merged, thanks! I'm glad to receive such insights from people with access to an entirely different class of hardware than I have. :smile:

File: src/heretic/model.py [hunk 1]
```diff
@@ -803,8 +803,10 @@ def hook(
 
                 # inputs[0] and outputs have shape (prompt, position, component),
                 # so this extracts the input/output at the end of each prompt.
-                input = inputs[0][:, -1, :].detach().clone()
-                output = outputs[:, -1, :].detach().clone()
+                # Move to CPU to decouple from device assignments, which can
+                # change between model reloads in multi-GPU configurations.
+                input = inputs[0][:, -1, :].detach().clone().cpu()
+                output = outputs[:, -1, :].detach().clone().cpu()
```
[the remark this answers is not in the document]
Comment (joninco) [reply src/heretic/model.py:809]: > medium
> 
> The `.clone()` call is redundant here. `tensor.cpu()` already creates a copy of the tensor data in CPU memory if the tensor is on a different device. You can simplify `.detach().clone().cpu()` to `.detach().cpu()` to avoid an unnecessary intermediate copy on the GPU.

Btw, Gemini is wrong. Without clone(), the "detached" tensor still points to the same GPU memory. The model would be unable to free that layer's activations because something still references them.
